src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -154,11 +139,7 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
     public ResponseEntity<Void> reset(@PathVariable Long id) {
         log.debug("REST request to reset Exercise : {}", id);
         Exercise exercise = exerciseService.findOneLoadParticipations(id);
-        Course course = exercise.getCourse();
-        User user = userService.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isInstructorInCourse(course, user) && !authCheckService.isAdmin()) {
-            return forbidden();
-        }
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) return forbidden();
```

Comment:
Please be more careful when changing existing code that works. This should only be possible for instructors, so change it to `isAtLeastInstructorForExercise`